apps/extension/src/routes/popup/approval/origin.tsx

```diff
@@ -0,0 +1,77 @@
+import { FadeTransition } from '@penumbra-zone/ui';
+import { useStore } from '../../../state';
+import { originApprovalSelector } from '../../../state/origin-approval';
+import { ApproveDeny } from './approve-deny';
+import { LinkGradientIcon } from '../../../icons';
+import { ExclamationTriangleIcon } from '@radix-ui/react-icons';
+
+export const OriginApproval = () => {
+  const { requestOrigin, favIconUrl, title, setAttitude, sendResponse } =
+    useStore(originApprovalSelector);
+
+  const approve = () => {
+    setAttitude(true);
+    sendResponse();
+    window.close();
+  };
+
+  const deny = () => {
+    setAttitude(false);
+    sendResponse();
+    window.close();
+  };
+
+  if (!requestOrigin) return null;
+
+  const originUrl = new URL(requestOrigin);
+  if (originUrl.origin !== requestOrigin) throw new Error('Invalid origin');
+  const { protocol, hostname, port } = originUrl;
+
+  return (
+    <FadeTransition>
+      <div className='flex min-h-[100vh] w-[100vw] flex-col gap-6'>
+        <h1 className='flex h-[70px] items-center justify-center border-b border-border-secondary font-headline text-xl font-semibold leading-[30px]'>
+          Connect
+        </h1>
+        <div className='mx-auto size-20'>
+          <LinkGradientIcon />
+        </div>
+        <div className='flex flex-1 flex-col items-start justify-between px-[30px] pb-[30px]'>
+          <div className='flex w-full flex-col gap-4'>
+            <div className='flex flex-col gap-2'>
+              <div className='flex h-11 w-full items-center rounded-lg border bg-background px-3 py-2 text-muted-foreground'>
+                {favIconUrl ? <img src={favIconUrl} alt='icon' className='h-11' /> : null}
```

Comment:
The favicon appears to exceed the size of the box. Can you make it the height (minus padding) of the box?

Reply:
yes. this was a deliberate aesthetic choice